mtest/operators_test.go

```diff
@@ -4,15 +4,204 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io/ioutil"
 	"strings"
 	"time"
 
 	"github.com/cybozu-go/cke"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func waitRebootCompletion(cluster *cke.Cluster) {
+	ts := time.Now()
+	Eventually(func() error {
+		var entries []*cke.RebootQueueEntry
+		data := ckecliSafe("reboot-queue", "list")
+		err := json.Unmarshal(data, &entries)
+		if err != nil {
+			return err
+		}
+		if len(entries) != 0 {
+			return fmt.Errorf("reboot entry is remaining")
+		}
+		return checkCluster(cluster, ts)
+	}).Should(Succeed())
+}
+
+func testRebootOperations(cluster *cke.Cluster) {
+	By("Rebooting nodes")
+	// this will run:
+	// - RebootOp
+	// - RebootDequeueOp
+	rebootTargets := node1
+	_, _, err := ckecliWithInput([]byte(rebootTargets), "reboot-queue", "add", "-")
+	Expect(err).ShouldNot(HaveOccurred())
+	rebootTargets = node2 + "\n" + node3
+	_, _, err = ckecliWithInput([]byte(rebootTargets), "reboot-queue", "add", "-")
+	Expect(err).ShouldNot(HaveOccurred())
+	waitRebootCompletion(cluster)
+
+	out, stderr, err := execAtWithInput(host1, ckecliSafe("history", "-n", "2"), "jq", "-s", ".")
+	Expect(err).ShouldNot(HaveOccurred(), "stderr: %s", stderr)
+	var history []*cke.Record
+	err = json.Unmarshal(out, &history)
+	Expect(err).ShouldNot(HaveOccurred())
+	Expect(history).Should(HaveLen(2))
+	Expect(history[0].Operation).To(Equal("rebootDequeue"))
+	Expect(history[1].Operation).To(Equal("reboot"))
+	Expect(history[1].Targets).Should(HaveLen(2))
+
+	By("Reboot operation gives up waiting node startup if deadline is exceeded")
+	previousCommand := cluster.Reboot.Command
+	cluster.Reboot.Command = []string{"sleep", "3600"}
+	_, err = ckecliClusterSet(cluster)
+	// wait for the previous reconciliation to be done
+	time.Sleep(time.Second * 3)
+
+	Expect(err).ShouldNot(HaveOccurred())
+	rebootTargets = node1
+	ckecliWithInput([]byte(rebootTargets), "reboot-queue", "add", "-")
+	waitRebootCompletion(cluster)
+
+	out, stderr, err = execAtWithInput(host1, ckecliSafe("history", "-n", "2"), "jq", "-s", ".")
```

Comment:
Instead of checking the error message, why don't you check the elapsed time for the timeout?